### ISSUE TITLE
Automate publishing

### ISSUE THREAD

**Issue by tfausak** [tfausak/haskell-knowledge-base#3]
I asked about this in the Obsidian Discord: https://discord.com/channels/686053708261228577/768134314864017429/833752684909428746

I don't think it's officially possible, but the API seems simple enough. I'll see if I can get a proof of concept working soon.

**Comment by tfausak** [comment]
It seems relatively easy to add and update files. You just need to hit an API endpoint like so:

``` sh
$ http -v post https://publish-01.obsidian.md/api/upload \
  obs-token:'...' \
  obs-id:'...' \
  Content-Type:application/octet-stream \
  obs-path:some-file \
  obs-hash:(sha256sum some-file | cut -d ' ' -f 1) \
  < some-file
```

**Comment by tfausak** [comment]
Removing files can be done with:

``` sh
$ http -v post https://publish-01.obsidian.md/api/remove \
  token='...' \
  id='...' \
  path=some-file
```

**Comment by tfausak** [comment]
You can get a list of known files and their hashes with the following API call:

``` sh
$ http -v post https://publish-01.obsidian.md/api/list \
  token='...' \
  id='...'
```

It returns a list like this:

``` json
[
    {
        "ctime": 1618795536330,
        "hash": "30941b080a45de244b1e5da1331f7476e4ce49cb66439a493b6289efbdb85c9b",
        "mtime": 1618866370468,
        "path": "README.md",
        "size": 824
    },
```

That could be used to diff the local files with the remote manifest to see what needs to be added, updated, or removed.

**Comment by tfausak** [comment]
Fixed by ced6e2236ff21cadb4106bdddc02654689770278.
